botocore/loaders.py

```diff
@@ -302,19 +374,18 @@ def list_available_services(self, type_name):
             # by searching for the corresponding type_name in each
             # potential directory.
             possible_services = [
-                d
-                for d in os.listdir(possible_path)
-                if os.path.isdir(os.path.join(possible_path, d))
+                path_obj
+                for path_obj in possible_path.iterdir()
+                if path_obj.is_dir()
             ]
-            for service_name in possible_services:
-                full_dirname = os.path.join(possible_path, service_name)
-                api_versions = os.listdir(full_dirname)
-                for api_version in api_versions:
-                    full_load_path = os.path.join(
-                        full_dirname, api_version, type_name
-                    )
-                    if self.file_loader.exists(full_load_path):
-                        services.add(service_name)
+            for service_name_path in possible_services:
+                full_load_paths = [
+                    path_obj.joinpath(type_name)
```

Comment:
Is this actually grabbing all the versions of the APIs with iterdir?

Reply:
Yes. Here is an example for `ec2` from my machine with the first part of the directory path redacted
```python
type_name = 'service-2'
service_path = `PosixPath('..../botocore/botocore/data/ec2')
[p.joinpath('service-2') for p in service_path.iterdir()]
[PosixPath('..../botocore/botocore/data/ec2/2015-10-01/service-2'),
 PosixPath('..../botocore/botocore/data/ec2/2016-11-15/service-2'),
 PosixPath('....botocore/botocore/data/ec2/2015-04-15/service-2'),
 PosixPath('..../botocore/botocore/data/ec2/2016-04-01/service-2'),
 PosixPath('..../botocore/botocore/data/ec2/2014-10-01/service-2'),
 PosixPath('..../botocore/botocore/data/ec2/2016-09-15/service-2'),
 PosixPath('..../botocore/botocore/data/ec2/2014-09-01/service-2'),
 PosixPath('..../botocore/botocore/data/ec2/2015-03-01/service-2')]